storage/uploader.go

```diff
@@ -0,0 +1,353 @@
+package storage
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+
+	authhttp "github.com/bsv-blockchain/go-sdk/auth/clients/authhttp"
+)
+
+// StorageUploader implements the StorageUploaderInterface
+type StorageUploader struct {
```

Comment:
Maybe just `Uploader` so it's `storage.Uploader` (same with `storage.StorageDownloader`)